### PR TITLE
Update to CLDR 44 and ICU 74

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Review by sffc — APPROVED** [review]
Probably need to update the testdata for segmenter and collator

File: components/segmenter/tests/testdata/SentenceBreakExtraTest.txt [hunk 1]
```diff
@@ -21,112 +21,4 @@
 ÷ 0021 × 0022 × 0020 ÷ 0022 ÷ # [STerm] [Close] [Sp] [Close]
 ÷ 0021 × 0022 × 0020 ÷ 0031 ÷ # [STerm] [Close] [Sp] [Numeric]
 ÷ 0021 × 0022 × 0020 ÷ 0041 ÷ # [STerm] [Close] [Sp] [Upper]
-÷ 0021 × 0022 × 0020 ÷ 3041 ÷ # [STerm] [Close] [Sp] [OLetter]
-# The following 100 monkey tests were generated from
```
Comment (robertbastian) [inline components/segmenter/tests/testdata/SentenceBreakExtraTest.txt:25]: @eggrobin these are breaking with ICU74, is it okay to remove them for now?

File: provider/adapters/tests/data/blob.postcard
Comment (robertbastian) [inline provider/adapters/tests/data/blob.postcard:1]: this is a fallback data size increase

**Review by sffc — APPROVED** [review]

**Comment by echeran** [comment]
For posterity searching's sake: this release (via ICU 74) includes Unicode 15.1 data.

**Comment by hsivonen** [comment]
@robertbastian Was it intentional to replace the placeholder riwords.txt with the real thing in this changeset (increasing the repo and released crate size with more test data)?

https://github.com/unicode-org/icu4x/blame/main/components/collator/tests/data/riwords.txt

**Comment by robertbastian** [comment]
I do not remember. I think the repo ship has long sailed (we have ~80MB of JSON test data), but you can exclude it from the crate without removing it from the repo if it's an issue.

**Pull request merged**